### DISCUSSION TITLE
Cannot run windows 11 or 10 ARM on UTM SE for iOS

### DISCUSSION THREAD

**Discussion by homurahomu1** [utmapp/UTM#3419] [UTM Help (iOS)] [answered]
Recently I tried to boot the windows 11 arm and windows 10 arm isos and VHDXs in the same way that you would do on Mac UTM, but it does not work. It just shows the "press any key to start from CD" and when I press a key it just stays on the UTM logo without any loading occurring. Trying the same procedure on my M1Pro Mac, it immediately boots into the installer for Windows 11 ARM after I press the key. Is there any way to fix this? thank you!

**Comment by conath** [comment, marked as answer]
Edit: solution from replies:

> The most common missing part to get Windows for ARM booting is to set the QEMU Machine Properties to highmem=off,virtualization=on. This is in the System tab under Advanced Settings.

> Another config to check in the System => Advanced section: CPU should be cortex-a72 and cores set to 2 with "force multicore" on.

Previous answer:

There is a technical reason for this and it's not in fact a problem with UTM.

On your M1 Pro Mac, the VM will run almost at the same speed as if you had installed Windows directly (without using VM software, which isn't currently possible).

UTM SE works *very* differently. It's called *Slow Edition* for a reason: in this version the entire virtual computer is **emulated**, meaning the iOS device must simulate every single piece of hardware including the CPU. This is extremely inefficient when compared to running in hypervisor (which is what your Mac does) and results in a ~90%+ speed reduction (guessing, compared to your M1 Pro).

So what you're seeing is just the Windows on ARM installer taking at least 10x longer to start up than normal. This can't be fixed by UTM because on iOS, Apple choose to deny both the hypervisor feature and the faster-emulation mode known as JIT. See #2271 for some more info.

**Reply by homurahomu1** [reply]
Hi! 
Thank you for taking your time explaining this to me, but there’s one thing that I didn’t explain properly. The VM literally does not start up no matter how long I wait. I’ve left it for 10 hours+ on the ARM windows booting the ISO and there is no loading icon. With x86_64 emulation using the x64 ISO it does boot into the windows installer in around 10 minutes or so, so I’m confused why the ARM version of windows(I tried multiple ISOs, from uupdump, the VHDX from Microsoft website) all just show a UTM logo after “press any key to boot to CD” without any sign of loading after 10+ hours. I’ve tried using Ubuntu Server ARM as well and it also does not work. My device I’m trying it on is an iPad Air 3 with the Apple A12 processor. Thanks if you can help me fix this problem, I understand that the entire dev team work really hard on making an app that no other dev team even dares to try, thank you for your effort, and if you manage to help me fix the issue I’d be extremely happy! :D

**Reply by conath** [reply]
This sounds like a configuration issue. The most common missing part to get Windows for ARM booting is to set the QEMU Machine Properties to `highmem=off,virtualization=on`. This is in the System tab under Advanced Settings.

Another config to check in the System => Advanced section:  CPU should be `cortex-a72` and cores set to 2 with "force multicore" on.

**Reply by homurahomu1** [reply]
Thank you sooooo much, virtualization=on fixed everything!

:D

**Comment by startergo** [comment]
@conath This is very confusing:
>This sounds like a configuration issue. The most common missing part to get Windows for ARM booting is to set the QEMU Machine Properties to highmem=off,virtualization=on. This is in the System tab under Advanced Settings.

Are you referring to the ARM on Macs or on iPadOS/iOS ?
